File: 1737-MaximumNestingDepthOfTheParentheses/1737-MaximumNestingDepthOfTheParentheses.py
# Last updated: 3/31/2026, 9:34:05 PM
class Solution:
    def maxDepth(self, s: str) -> int:
        maxdepth = 0
        depth = 0
        for i in s:
            if i == "(":
                depth +=1
                maxdepth = max(depth, maxdepth)
            elif i==")":
                depth-=1
        return maxdepth

File: 1737-MaximumNestingDepthOfTheParentheses/1737-MaximumNestingDepthOfTheParentheses.cpp
// Last updated: 3/31/2026, 9:33:59 PM
class Solution {
public:
    int maxDepth(string s) {
        int maxd = 0, d= 0;
        for (auto i: s){
            if (i =='('){
                d+=1;
                maxd = max(d,maxd);
            }
            else if (i==')'){
                d-=1;
            }
        }
        return maxd;
    }
};